[Plugin] Add documentation for the VCR plugin

GaryPEGEOT opened the pull request:
See php-http/vcr-plugin#3

dbu commented on plugins/vcr.rst:
```diff
@@ -0,0 +1,96 @@
+VCR Plugin
```
i'd love the title to say in 2 words what it does, for those not making the association with video cassette recorders (feeling old saying this, but people starting to code might only have seen such a device in a museum :-D). that title will show up in the table of contents and should already indicate what the plugin does.

something like "VCR Plugin - Record and Replay Responses"

dbu commented on plugins/vcr.rst:
```diff
@@ -0,0 +1,96 @@
+VCR Plugin
+==========
+
+The VCR plugins allow you to record & replay HTTP responses.
```
should we add "This is mainly useful for testing purposes."? its obvious to you and me, but maybe not to everyone.

dbu commented on plugins/vcr.rst:
```diff
@@ -0,0 +1,96 @@
+The identifier must be safe to use with a filesystem.
+The plugin provide a default naming strategy, the ``PathNamingStrategy``. You can define two options:
+
+* **hash_headers**: the list of header names to hash (Ex: Authorization),
```
the hashing is a implementation detail. the main thing is that these headers make requests distinct. and we should be clear that while we specify the "header name", we hash the "header", not just the "header name"

dbu commented on plugins/vcr.rst:
```diff
@@ -0,0 +1,96 @@
+The plugin provide a default naming strategy, the ``PathNamingStrategy``. You can define two options:
+
+* **hash_headers**: the list of header names to hash (Ex: Authorization),
+* **hash_body_methods**: methods for which the body will be hashed (Default: PUT, POST, PATCH)
```
similar here, the config tells for which request methods the body makes requests distinct.

dbu commented on plugins/vcr.rst:
```diff
@@ -0,0 +1,96 @@
+The recorder
+************
+
+The recorder records and replays responses. The plugin provide a two recorders:
```
s/provide a two/provides two/

dbu commented on plugins/vcr.rst:
```diff
@@ -0,0 +1,96 @@
+The plugins
+***********
+
+VCR come built-in with two plugins:
```
i'd say "There are two plugins, one to record responses, the other to replay them."

dbu commented on plugins/vcr.rst:
```diff
@@ -0,0 +1,96 @@
+* ``Http\Client\Plugin\Vcr\RecordPlugin``, use a ``RecorderInterface`` instance to record the responses,
+
+If you plan on using both plugins at the same time (Replay or Record), the ``ReplayPlugin`` **must always** come first.
+Please also note that by default, the ``ReplayPlugin`` throw an exception when it cannot replay a request, unless you set the third constructor argument to ``false`` (See example below).
```
s/throw/throws/

and i'd add a dot in between the 2 parts. "... replay a request, unless ..." => "... replay a request. If you want the plugin to continue the request (possibly to the actual server), set the third constructor argument..."

dbu commented on plugins/vcr.rst:
```diff
@@ -0,0 +1,96 @@
+    );
+
+    // Will be recorded in "some/dir/in/vcs"
+    $client->sendRequest(//...);
```
i'd use an actual url here to make it clear you call the same url twice. you could use httplug.io.

GaryPEGEOT commented:
I've addressed all the comments (unless I forgot one), also added a comment about the responses headers added by both plugins. Might need some proof reading :smile:

GaryPEGEOT commented on plugins/vcr.rst:
```diff
@@ -22,8 +25,8 @@ The naming strategy turn a request into a deterministic and unique identifier.
 
-* **hash_headers**: the list of header names to hash (Ex: Authorization),
-* **hash_body_methods**: methods for which the body will be hashed (Default: PUT, POST, PATCH)
+* **hash_headers**: the list of header(s) that make the request unique (Ex: 'Authorization'). The content of the header will be hashed to generate a unique signature. By default no header is used.
```
Yeah [both are hashed](https://github.com/php-http/vcr-plugin/blob/master/src/NamingStrategy/PathNamingStrategy.php#L84), you want me to re-phrase?

dbu replied:
thanks!

dbu commented:
can you please rebase on master? there was some change in the way we check the doc that made ci failing, i fixed it on master.

GaryPEGEOT commented:
> can you please rebase on master? there was some change in the way we check the doc that made ci failing, i fixed it on master.

Done!

dbu commented:
awesome, cheers!
